sockets/unix_socket_test_unix.go

```diff
@@ -0,0 +1,34 @@
+//go:build !windows
+// +build !windows
+
+package sockets
+
+import (
+	"os"
+	"syscall"
+	"testing"
+)
+
+func createTestUnixSocket(t *testing.T) (listener net.Listener, path string) {
+	uid, gid := os.Getuid(), os.Getgid()
+	perms := os.FileMode(0660)
+	path = "/tmp/test.sock"
```

Comment:
Should be created in `t.TempDir()`

Reply:
This was that way even before my PR, so technically it is unrelated. But I fixed anyway.